### PR TITLE
OPEN-194: load-csv: create missing datasets, remove org extras

### PR DESCRIPTION
**Pull request by RabiaSajjad**

### CODE REVIEW

File: ckanext/recombinant/commands.py [hunk 1]
```diff
@@ -233,35 +236,47 @@ def _delete(self, dataset_types):
...
         assert csv_name.endswith('.csv'), csv_name
         resource_name = csv_name[:-4]
         print resource_name
         chromo = get_chromo(resource_name)
+
+        if lenient:
+            # remove 'csv_org_extras' fields
+            if 'csv_org_extras' in chromo:
+                extra_fields = chromo['csv_org_extras']
+                del chromo['csv_org_extras']
```
Comment (wardi) [inline ckanext/recombinant/commands.py]: this doesn't affect anything AFAICT

File: ckanext/recombinant/commands.py [hunk 2]
```diff
@@ -282,6 +297,14 @@ def _load_one_csv_file(self, name):
                             r[k] = r[k].split(',')
 
             print '-', org_name, len(records)
+
+            if lenient:
```
Comment (wardi) [inline ckanext/recombinant/commands.py]: how about we remove the flag and just say `if extra_fields` here

**Comment by wardi** [comment]
The lenient flag can be removed now too

**Comment by RabiaSajjad** [comment]
Should I default `strict=False`?
https://github.com/open-data/ckanext-recombinant/blob/60cba2e7892ce3d01f444e29c1a7a2da9dfc0cd8/ckanext/recombinant/commands.py#L256

**Comment by wardi** [comment]
Actually, that question points to `csv_data_batch` as a better place to remove the `csv_org_extras` columns.

**Comment by RabiaSajjad** [comment]
looks like `csv_data_batch` taking care of `csv_org_extras` in 
https://github.com/open-data/ckanext-recombinant/blob/60cba2e7892ce3d01f444e29c1a7a2da9dfc0cd8/ckanext/recombinant/read_csv.py#L29

**Comment by wardi** [comment]
No that just prevents an asssert failure when strict=True

**Comment by RabiaSajjad** [comment]
Hi Ian, I think `strict` is `True` by default

https://github.com/open-data/ckanext-recombinant/blob/ed9d723fa125e6e697a75e9df4a1e4c11458686a/ckanext/recombinant/commands.py#L253

Moreover, to test I started over again, delete ati, load the csv as-is, update the fields, migrate the csv and load it back. All completed with no issues. Let me know if you think further changes are required.

Thanks!

**Pull request merged**